storybookjs/storybook
# Docs: Raw support for `mjs`, `jsx`, and `tsx` files

## What I did

Enabled support for loading truly raw mjs,jsx, and tsx files by disabling the babel-loader when the raw query is set.

## How to test

1. Run a sandbox for template, e.g. `yarn task --task sandbox --template react-webpack/18-ts`
2. Open Storybook in your browser
3. Comment the webpackFinal overrides in .storybook/main.ts
4. Access ?path=/docs/addons-docs-stories-mdx-raw-source--docs
5. Verify you see:
```jsx
import React, { Fragment } from 'react';

export const MY_VALUE_I_WANT_TO_SHOW_IN_STORY = 'Hello World!';

export const RawSource = () => {
  return <Fragment>{MY_VALUE_I_WANT_TO_SHOW_IN_STORY}</Fragment>;
};
```
Instead of:
```jsx
import React, { Fragment } from "react";
export const MY_VALUE_I_WANT_TO_SHOW_IN_STORY = "Hello World!";
export const RawSource = () => {
  return /* @__PURE__ */ React.createElement(Fragment, null, MY_VALUE_I_WANT_TO_SHOW_IN_STORY);
};
```

## Checklist



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

      make sure to add the `ci:merged` or `ci:daily` GH label to it.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`
